components/bl-dashlet/src/helpers.js

```diff
@@ -124,3 +124,98 @@ export const getPosition = (ref, coords) => {
     ),
   };
 };
+
+export const useContextMenuPositionHandler = (contextMenuRef, isMenuOpen) => {
+  const [sides, setSides] = useState({ left: false, top: false });
+  const [isChecked, setIsChecked] = useState(false);
+
+  useEffect(() => {
+    if (contextMenuRef.current && isMenuOpen) {
+      contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+
+      window.addEventListener('resize', () => {
+        contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+      });
+
+    } else {
+      setSides({ left: false, top: false });
+      setIsChecked(false);
+    }
+
+    return () => {
+      window.removeEventListener('resize', () => {
+        contextMenuPosition(contextMenuRef, isMenuOpen, sides, setIsChecked, setSides);
+      });
+    };
+  }, [contextMenuRef, isMenuOpen]);
+
+  return { sides, isChecked };
+};
+
+const contextMenuPosition = (contextMenuRef, isMenuOpen, sides, setIsChecked, setSides) => {
```

Comment:
contextMenuPosition - could be right, left, top and so on.

This variable name doesn't much thing that function do.

checkOverflow / prepareToShow/ handleOverflow will be much better